cmd/snap/error.go

```diff
@@ -190,3 +220,136 @@ If you understand and want to proceed repeat the command including --classic.
 
 	return msg, nil
 }
+
+func snapRevisionNotAvailableMessage(kind, snapName, action, arch, channel string, releases []interface{}) string {
+	req, err := snap.ParseChannel(channel, arch)
+	if err != nil {
+		msg := fmt.Sprintf(i18n.G("requested what looks like an invalid channel for snap %[1]q.\nPlease use 'snap info %[1]s' to list available releases."), snapName)
+		return msg
+	}
+	avail := make([]*snap.Channel, 0, len(releases))
+	for _, v := range releases {
+		rel, _ := v.(map[string]interface{})
+		relCh, _ := rel["channel"].(string)
+		relArch, _ := rel["architecture"].(string)
+		if relArch == "" {
+			logger.Debugf("internal error: revision-not-found store error carries a release with invalid/empty architecture: %v", v)
+			continue
+		}
+		a, err := snap.ParseChannel(relCh, relArch)
+		if err != nil {
+			logger.Debugf("internal error: revision-not-found store error carries a release with invalid/empty channel (%v): %v", err, v)
+			continue
+		}
+		avail = append(avail, &a)
+	}
+
+	matches := map[string][]*snap.Channel{}
+	for _, a := range avail {
+		m := req.Match(a)
+		matchRepr := m.String()
+		if matchRepr != "" {
+			matches[matchRepr] = append(matches[matchRepr], a)
+		}
+	}
+
+	if kind == client.ErrorKindRevisionNotAvailableForArchitecture {
+		// TODO: add "Get more information..." hints once snap info
+		// support showing multiple/all archs
+
+		if hits := matches["track:risk"]; len(hits) != 0 {
+			archs := strings.Join(archsForChannels(hits), ", ")
+			msg := fmt.Sprintf(i18n.G("snap %q is not available on %v for this architecture (%s) but exists on other architectures (%s)."), snapName, req, arch, archs)
```

Comment:
a `// TRANSLATORS` comment to help them figure out what all the `%`s are, please